service-api/module/Application/src/Experian/Crosscore/FraudApi/FraudApiService.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Experian\Crosscore\FraudApi;
+
+use Application\Experian\Crosscore\AuthApi\AuthApiException;
+use Application\Experian\Crosscore\AuthApi\AuthApiService;
+use Application\Experian\Crosscore\FraudApi\DTO\RequestDTO;
+use Application\Experian\Crosscore\FraudApi\DTO\ResponseDTO;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use Laminas\Http\Response;
+use Ramsey\Uuid\Uuid;
+
+class FraudApiService
+{
+    public function __construct(
+        private readonly Client $client,
+        private readonly AuthApiService $experianCrosscoreAuthApiService,
+        private readonly array $config
+    ) {
+    }
+
+    public int $authCount = 0;
+
+    /**
+     * @throws GuzzleException
+     * @throws AuthApiException
+     */
+    public function makeHeaders(): array
+    {
+        return [
+            'Content-Type' => 'application/json',
+            'Authorization' => $this->experianCrosscoreAuthApiService->retrieveCachedTokenResponse(),
+            'X-User-Domain' => $this->config['domain']
+        ];
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws FraudApiException
+     */
+    public function getFraudScore(
+        RequestDTO $experianCrosscoreFraudRequestDTO
+    ): array {
+        $response = $this->makeRequest($experianCrosscoreFraudRequestDTO);
+        return $response->toArray();
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws FraudApiException
+     * @psalm-suppress InvalidReturnType
+     */
+    public function makeRequest(
+        RequestDTO $experianCrosscoreFraudRequestDTO
+    ): ResponseDTO {
+        $this->authCount++;
+        try {
+            $postBody = $this->constructRequestBody($experianCrosscoreFraudRequestDTO);
+
+            $response = $this->client->request(
+                'POST',
+                '/3',
+                [
+                    'headers' => $this->makeHeaders(),
+                    'json' => json_encode($postBody)
+                ]
+            );
+
+            $responseArray = json_decode($response->getBody()->getContents(), true);
+
+            return new ResponseDTO(
+                $responseArray
+            );
+        } catch (ClientException $clientException) {
+            if ($clientException->getResponse()->getStatusCode() == Response::STATUS_CODE_401) {
+                $this->experianCrosscoreAuthApiService->authenticate();
+                $this->makeRequest($experianCrosscoreFraudRequestDTO);
+            } elseif ($this->authCount > 2) {
+                throw $clientException;
+            } else {
```

Comment:
Need to check `authCount` in the initial if statement or the first one will always trigger:
```suggestion
            if ($clientException->getResponse()->getStatusCode() == Response::STATUS_CODE_401 && $this->authCount < 2) {
                $this->experianCrosscoreAuthApiService->authenticate();
                $this->makeRequest($experianCrosscoreFraudRequestDTO);
            } else {
```